packages/ui-client/src/hooks/useFeaturePreviewList.ts

```diff
@@ -82,22 +87,46 @@ export const defaultFeaturesPreview: FeaturePreviewProps[] = [
 
 export const enabledDefaultFeatures = defaultFeaturesPreview.filter((feature) => feature.enabled);
 
-export const useFeaturePreviewList = () => {
+export const usePreferenceFeaturePreviewList = () => {
 	const featurePreviewEnabled = useSetting<boolean>('Accounts_AllowFeaturePreview');
-	const userFeaturesPreview = useUserPreference<FeaturePreviewProps[]>('featuresPreview');
+	const userFeaturesPreviewPreference = useUserPreference<FeaturePreviewProps[]>('featuresPreview');
+	// TODO: Remove this logic after we have a way to store object settings.
+	const userFeaturesPreview = useMemo<FeaturePreviewProps[]>(() => {
+		if (typeof userFeaturesPreviewPreference === 'string') {
+			return JSON.parse(userFeaturesPreviewPreference);
+		}
+		return userFeaturesPreviewPreference;
+	}, [userFeaturesPreviewPreference]);
+	const { unseenFeatures, features } = useFeaturePreviewList(userFeaturesPreview);
 
 	if (!featurePreviewEnabled) {
 		return { unseenFeatures: 0, features: [] as FeaturePreviewProps[], featurePreviewEnabled };
 	}
+	return { unseenFeatures, features, featurePreviewEnabled };
+};
+
+export const useDefaultSettingFeaturePreviewList = () => {
+	const featurePreviewSettingJSON = useSetting<string>('Accounts_Default_User_Preferences_featuresPreview');
+	// TODO: Remove this logic after we have a way to store object settings.
+	const settingFeaturePreview = useMemo<FeaturePreviewProps[]>(() => {
+		if (typeof featurePreviewSettingJSON === 'string') {
+			return JSON.parse(featurePreviewSettingJSON);
+		}
+		return featurePreviewSettingJSON;
+	}, [featurePreviewSettingJSON]);
+
+	return useFeaturePreviewList(settingFeaturePreview);
+};
```

Comment:
This is (with some smol changes) the same code as L94. Can we make this a func and call it in both places? (even if its a local func, no need for new files/helpers/etc)

Reply:
Agreed. No need to make a helper as we don't want to encourage this type of patter. Just abstract to a local function.